### PR TITLE
Style notifications screen

### LINKED ISSUE #99: implement `notifications_view` UI

### PR DESCRIPTION
**Pull request by milenark**
Notification view more aligned with figma design: 

Screenshot 2023-12-18 at 13 07 40
Screenshot 2023-12-18 at 13 10 37

### CODE REVIEW

**Review by Anishyou — COMMENTED** [review]
Looks good to me!

**Review by Anishyou — COMMENTED** [review]
I would suggest , use themes and constants. It is better for future purposes.
then we can merge

**Comment by carlobortolan** [comment]
Any updates @milenark ?

File: lib/base/networking/api/handler/notification_handler.dart [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    return _grpcHandler.callGrpcMethod(
+      (client) async {
+        final response = await client
+            .getFeatureNotifications(GetFeatureNotificationsRequest());
+        _logger.d('Public courses: ${response.featureNotifications}');
+        return response.featureNotifications;
+      },
+    );
+  }
+
```
Comment (GravityDarkLab) [inline lib/base/networking/api/handler/notification_handler.dart:25]:
```suggestion
/// Fetches Banner Alerts.
  ///
  /// This method sends a `getBannerAlerts` gRPC call to fetch the banner alerts.
  ///
  /// returns a [List<BannerAlert>] instance that represents the banner alerts.
```

File: lib/base/networking/api/handler/notification_handler.dart [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+/// Handles Notification-related data operations.
+///
+/// This class is responsible for fetching and posting notification-related data, such as fetching feature Notification and banner Alerts.
+class NotificationHandler {
+  static final Logger _logger = Logger();
+  final GrpcHandler _grpcHandler;
+
+  NotificationHandler(this._grpcHandler);
+
```
Comment (GravityDarkLab) [inline lib/base/networking/api/handler/notification_handler.dart:13]:
```suggestion
  /// Fetches featured notifications.
  ///
  /// This method sends a `getFeatureNotifications` gRPC call to fetch the featured notifications.
  ///
  /// returns a [List<FeatureNotification>] instance that represents the featured notifications.
```

File: lib/views/components/custom_bottom_nav_bar.dart [hunk 3]
```diff
@@ -2,11 +2,11 @@ import 'package:flutter/material.dart';
 import 'package:flutter_riverpod/flutter_riverpod.dart';
 import 'package:gocast_mobile/providers.dart';
 import 'package:gocast_mobile/views/course_view/pinned_courses_view/pinned_courses_view.dart';
-import 'package:gocast_mobile/views/notifications_view/notifications_screen_view.dart';
-
 import 'package:gocast_mobile/views/course_view/courses_overview.dart';
 import 'package:gocast_mobile/views/course_view/downloaded_courses_view/downloaded_courses_view.dart';
 
+import '../notifications_view/notifications_overview.dart';
```
Comment (GravityDarkLab) [inline lib/views/components/custom_bottom_nav_bar.dart:8]: change to `package:gocast_mobile..........`

File: lib/views/notifications_view/notifications_overview.dart [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    _fetchData();
+  }
+
+  void _fetchData() {
+    Future.delayed(Duration.zero, () async {
+      try {
+        await ref.read(userViewModelProvider.notifier).fetchFeatureNotifications();
+        await ref.read(userViewModelProvider.notifier).fetchBannerAlerts();
+      } catch (e) {
+        // Handle errors if needed
```
Comment (GravityDarkLab) [inline lib/views/notifications_view/notifications_overview.dart:27]: Handle Error Here

File: lib/views/notifications_view/notifications_screen_view.dart [hunk 5]
```diff
@@ -1,11 +1,26 @@
 import 'package:flutter/material.dart';
 import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:gocast_mobile/base/networking/api/gocast/api_v2.pb.dart';
 import 'package:gocast_mobile/views/components/base_view.dart';
 
+import '../../utils/constants.dart';
+import '../settings_view/settings_screen_view.dart';
+
```
Comment (GravityDarkLab) [inline lib/views/notifications_view/notifications_screen_view.dart:8]: use `package.....`

File: lib/view_models/user_view_model.dart [hunk 6]
```diff
@@ -12,6 +12,8 @@ import 'package:gocast_mobile/models/user/user_state_model.dart';
 import 'package:gocast_mobile/utils/globals.dart';
 import 'package:logger/logger.dart';
 
+import '../base/networking/api/handler/notification_handler.dart';
```
Comment (GravityDarkLab) [inline lib/view_models/user_view_model.dart:15]: Use `package:.....`

**Review by GravityDarkLab — CHANGES_REQUESTED** [review]

File: lib/view_models/user_view_model.dart (hunk 6 above)
Comment (carlobortolan) [reply lib/view_models/user_view_model.dart]: Done ✅

File: lib/views/notifications_view/notifications_screen_view.dart (hunk 5 above)
Comment (carlobortolan) [reply lib/views/notifications_view/notifications_screen_view.dart]: Done ✅

File: lib/views/components/custom_bottom_nav_bar.dart (hunk 3 above)
Comment (carlobortolan) [reply lib/views/components/custom_bottom_nav_bar.dart]: Done ✅

File: lib/base/networking/api/handler/notification_handler.dart (hunk 2 above)
Comment (carlobortolan) [reply lib/base/networking/api/handler/notification_handler.dart:13]: Done ✅

File: lib/base/networking/api/handler/notification_handler.dart (hunk 1 above)
Comment (carlobortolan) [reply lib/base/networking/api/handler/notification_handler.dart:30]: Done ✅

File: lib/views/notifications_view/notifications_overview.dart (hunk 4 above)
Comment (carlobortolan) [reply lib/views/notifications_view/notifications_overview.dart]: Done ✅ (Error is already handled in UserViewModel)

**Review by carlobortolan — DISMISSED** [review]

File: lib/views/components/custom_search_top_nav_bar.dart
Comment (GravityDarkLab) [inline lib/views/components/custom_search_top_nav_bar.dart:1]: Merge from dev once #144 is merged.

**Review by GravityDarkLab — CHANGES_REQUESTED** [review]
Merge from dev once #144 is merged.

**Review by GravityDarkLab — APPROVED** [review]
looks good to me

**Pull request merged**